### ISSUE TITLE
False-positive `inconsistent-mro` when subclassing `Generic` and a "protocol-like ABC"

### ISSUE THREAD

**Issue by gschaffner** [astral-sh/ty#2052]
### Summary

Reproducer ([playground](https://play.ty.dev/b69404c6-38c0-4969-b4b8-f3755b668c5a)):

```python
from contextlib import AbstractContextManager
from typing import Generic
from typing import TypeVar

T = TypeVar("T")


class A(Generic[T], AbstractContextManager):  # error: inconsistent-mro
    ...


class B(AbstractContextManager, Generic[T]):  # no error
    ...
```

Output:

```console
$ ty check eg.py
error[inconsistent-mro]: Cannot create a consistent method resolution order (MRO) for class `A` with bases list `[<special-form 'typing.Generic[T]'>, <class 'AbstractContextManager'>]`
 --> eg.py:8:1
  |
8 | / class A(Generic[T], AbstractContextManager):  # error: inconsistent-mro
9 | |     ...
  | |_______^
  |
info: rule `inconsistent-mro` is enabled by default

Found 1 diagnostic
```

https://github.com/python/typeshed/pull/13005 is related.

### Version

ty 0.0.3

**Comment by carljm** [comment]
Thanks for reporting this!

**Comment by carljm** [comment]
Closing this as not-planned per the analysis in https://github.com/astral-sh/ruff/pull/24619#issuecomment-4243077502

This is a true positive given the definitions in typeshed; it's only a false positive relative to runtime because the typeshed inheritance hierarchies don't look like the runtime ones. Following the recommended pattern and putting `Generic[T]` last fixes the issue.
